### PR TITLE
Plugin refactor

### PR DESCRIPTION
**Pull request by james-nesbitt**
This patch rewires the existing code to be based on 2 different plugin types:
1. ActionType: (new) Is created in code, as needed, and passed to the dispatcher. Has arbitrary constructor and properties/methods (add methods and properties that make sense.)
2. Handler : (legacy Statistics plugin) Is created by the dispatcher, and given an ActionType plugin instance.

With this approach:
- We deprecate the StatisticsItem class;
- We no longer have to decide what to do with StatisticsItem
- We no longer have to decide what fields to give StatisticsItem, each plugin can have it's own definition

Triggering code should decide what plugin it needs, and create instances directly using the manager, and pass the instances to the dispatcher.

This PR deprecates https://github.com/james-nesbitt/drupal-sapi2-prototype/pull/25 , which had more stuff, in fewer commits

### CODE REVIEW

**Comment by james-nesbitt** [comment]
Also: moved the plugin managers and interface to the root src folder, which was something previously discussed.

**Comment by james-nesbitt** [comment]
Updated to also include a new "tagged" action typue plugin

File: src/Controller/JsActionCaptureController.php [hunk 1]
```diff
@@ -19,20 +20,44 @@
...
 class JsActionCaptureController extends ControllerBase implements ContainerInjectionInterface {
 
-  /** @var \Symfony\Component\HttpFoundation\RequestStack $requestStack */
+  /**
+   * @var \Symfony\Component\HttpFoundation\RequestStack $requestStack
+   *   Used to retrieve POST variables to create Action data
+   */
   protected $requestStack;
-  /** @var \Drupal\sapi\DispatcherInterface $sapiDispatcher */
+  /**
```
Comment (guncha25) [inline src/Controller/JsActionCaptureController.php:28]: Missing space in between.

File: src/Controller/JsActionCaptureController.php [hunk 2]
```diff
@@ -19,20 +20,44 @@
...
+   */
+  protected $container;
+
+  /**
    * JsActionCaptureController constructor.
    *
    * @param \Symfony\Component\HttpFoundation\RequestStack $requestStack
    * @param \Drupal\sapi\DispatcherInterface $sapiDispatcher
+   * @param ContainerInterface $container
    */
```
Comment (guncha25) [inline src/Controller/JsActionCaptureController.php:55]: Missing @param PluginManagerInterface $sapi_action_type_manager

**Comment by james-nesbitt** [comment]
@guncha failed to see your comments before merging.  I will create a new PR for those fixes.  My bad.

**Pull request merged**